api/desecapi/models.py

```diff
@@ -204,6 +205,20 @@ def send_email(self, reason, context=None, recipient=None):
 ]
 
 
+class DomainManager(Manager):
+    def filter_qname(self, qname: str, **kwargs) -> Domain:
```

Comment:
I think this doesn't return a `Domain`, but some queryset. Although I rather see it return `Optional[Domain]`

Reply:
fixed (`models.query.QuerySet`). That's why I don't like annotations for now, as they become inconsistent easily as long as they aren't enforced.

Reply:
You can enforce them by observing PyCharm's warnings